packages/main/src/Table.js

```diff
@@ -541,6 +549,103 @@ class Table extends UI5Element {
 		if (isAltUp || isDownAlt(event)) {
 			return this._handleArrowAlt(isAltUp, event.target);
 		}
+
+		if ((isUpShift(event) || isDownShift(event)) && this.isMultiSelect) {
+			this._handleArrowNav(event);
+		}
+
+		if (isHomeCtrl(event)) {
+			this._itemNavigation._handleHome(event);
+			this._itemNavigation._applyTabIndex();
+			this._itemNavigation._focusCurrentItem();
+		}
+
+		if (isEndCtrl(event)) {
+			this._itemNavigation._handleEnd(event);
+			this._itemNavigation._applyTabIndex();
+			this._itemNavigation._focusCurrentItem();
+		}
+
+		if ((isHomeShift(event) || isEndShift(event)) && this.isMultiSelect) {
+			this._handleHomeEndSelection(event);
+		}
+	}
+
+	_handleArrowNav(event) {
+		const isRowFocused = this._getActiveElementTagName() === "ui5-table-row";
+
+		if (!isRowFocused) {
+			return;
+		}
+
+		const previouslySelectedRows = this.selectedRows;
+		const currentItem = this.currentItem;
+		const currentItemIdx = this.currentItemIdx;
+
+		const prevItemIdx = currentItemIdx - 1;
+		const nextItemIdx = currentItemIdx + 1;
+
+		const prevItem = this.rows[prevItemIdx];
+		const nextItem = this.rows[nextItemIdx];
+
+		if (isUpShift(event) && currentItem.selected) {
+			prevItem.selected = true;
+			prevItem.focus();
+		}
+
+		if (isUpShift(event) && !currentItem.selected && prevItem.selected) {
+			prevItem.selected = false;
+			prevItem.focus();
+		}
+
+		if (isDownShift(event) && currentItem.selected) {
+			nextItem.selected = true;
+			nextItem.focus();
+		}
+
+		if (isDownShift(event) && !currentItem.selected && nextItem.selected) {
```

Comment:
Same as above, except for `SHIFT`+`DOWN`:

```suggestion
		if (isDownShift(event) && !currentItem.selected) {
```